### PR TITLE
Feat/hipaa compliance

### PR DESCRIPTION
**Pull request by gnpthbalaji**
## What Changed

- Created and populated the initial documentation for the HIPAA-compliance skill.
- Added a comprehensive overview of the compliance requirements.
- Defined core concepts and specific triggers for when to activate HIPAA protocols within the workflow.

## Why This Change
To provide Claude with a clear, standardized reference for HIPAA compliance. This ensures that Claude or other Coding Agents analysts understand the legal and technical boundaries when handling Protected Health Information (PHI), reducing the risk of compliance violations during feature development.

## Testing Done

- [x] Manual testing completed
- [x] Automated tests pass locally (`node tests/run-all.js`)
- [ ] Edge cases considered and tested

## Type of Change
- [ ] `fix:` Bug fix
- [x] `feat:` New feature
- [ ] `refactor:` Code refactoring
- [ ] `docs:` Documentation
- [ ] `test:` Tests
- [ ] `chore:` Maintenance/tooling
- [ ] `ci:` CI/CD changes

## Security & Quality Checklist
- [x] No secrets or API keys committed (ghp_, sk-, AKIA, xoxb, xoxp patterns checked)
- [x] JSON files validate cleanly
- [x] Shell scripts pass shellcheck (if applicable)
- [x] Pre-commit hooks pass locally (if configured)
- [x] No sensitive data exposed in logs or output
- [x] Follows conventional commits format

## Documentation
- [x] Updated relevant documentation
- [x] Added comments for complex logic
- [ ] README updated (if needed)

---
## Summary by cubic
Adds the `hipaa-compliance` skill with a complete guidance doc to help teams design, build, and review features that handle PHI safely. It auto-activates in healthcare contexts, provides tiered block/warn rules, includes “How It Works” and examples, and clarifies log retention with a compliant app logging example.

- **New Features**
  - Activation rules and core concepts, including the 18 PHI identifiers.
  - Enforcement tiers (critical, warning, advisory) with block/warn guidance and build/code review checklists.
  - Architecture guidance, anti-patterns with safe alternatives, third-party/BAA + LLM safeguards, and data lifecycle; cross-links to `healthcare-phi-compliance`.
  - “How It Works” (identification, encryption TLS 1.2+/AES-256, RBAC, auditability) and “Examples” showing compliant vs. non-compliant analysis and logging.

- **Refactors**
  - Documentation refinements for clarity and consistency: renamed “When to Activate” to “When to Use,” removed redundant wording, refined a compliant example, and fixed backtick formatting in the “Non-Compliant” example.
  - Clarified audit log retention guidance (follow applicable laws/policies; avoid blanket six-year rule) and added an explicit compliant app logging example.

<sup>Written for commit 0eb308faa4ef41b6c194877251a7b62d7a0282bd. Summary will update on new commits.</sup>

## Summary by CodeRabbit

* **Documentation**
  * Added comprehensive HIPAA/PHI compliance guidance: definitions (PHI, covered entities, BAAs), the 18 PHI identifiers, and severity tiers (Critical/Warning/Advisory) with example failure modes.
  * Includes actionable checklists for new systems and code reviews, architecture and access-control recommendations, strict "never include PHI in logs" audit rules, an audit-log schema, PHI lifecycle table, third‑party/BAA verification guidance, and safe AI/LLM usage practices.

### CODE REVIEW

File: skills/hipaa-compliance/SKILL.md [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+## Core Concepts
+
+| Term                   | Definition                                                                                       |
+| ---------------------- | ------------------------------------------------------------------------------------------------ |
+| **PHI**                | Any data that can identify a patient — directly or indirectly — combined with health information |
+| **Covered Entity**     | Health plans, clearinghouses, and healthcare providers who transmit PHI electronically           |
+| **Business Associate** | Vendors/services that handle PHI on behalf of a covered entity (requires BAA)                    |
+| **BAA**                | Business Associate Agreement — contract required before sharing PHI with a third party           |
+| **Minimum Necessary**  | Only access or expose the least amount of PHI required for a given task                          |
+| **De-identification**  | Removing 18 HIPAA identifiers so data no longer qualifies as PHI                                 |
```
[the remark this answers is not in the document]
Comment (gnpthbalaji) [reply skills/hipaa-compliance/SKILL.md:38]: 18 is the official list from HIPAA act. 

- DHCS - https://www.dhcs.ca.gov/dataandstats/data/Pages/ListofHIPAAIdentifiers.aspx
- Berkeley - https://cphs.berkeley.edu/hipaa/hipaa18.html

File: skills/hipaa-compliance/SKILL.md [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+
+- **Non-Compliant:** A data analyst downloads a CSV containing patient names and heart rates to their local machine to run a Python script.
+- **Compliant:** The analyst uses a de-identified dataset that satisfies HIPAA Safe Harbor (all 18 identifiers removed, or expert determination documented), and performs analysis only in a HIPAA-eligible environment with a signed BAA and appropriate access/audit controls.
+
+### 2. Application Logging
+
+- **Non-Compliant:**
+  ```python
+  logger.info(f"User {patient_name} updated their prescription for {medication_name}")
+
```
[the remark this answers is not in the document]
Comment (gnpthbalaji) [reply skills/hipaa-compliance/SKILL.md:238]: Both non compliant and compliant are present for example 2 

### 2. Application Logging

- **Non-Compliant:** `python
logger.info(f"User {patient_name} updated their prescription for {medication_name}")`

- **Compliant:** `python
logger.info(f"User {user_id} updated a record in the 'Prescriptions' table")`
  _Note: The specific health data is kept in the encrypted database, not the plaintext application logs._

File: skills/hipaa-compliance/SKILL.md [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+- [ ] Internal service communication also encrypted (not just external)
+- [ ] File exports containing PHI encrypted before delivery
+
+**Audit Logging**
+
+- [ ] Every PHI read, write, update, and delete is logged
+- [ ] Log records include: user/service ID, action, resource ID, timestamp
+- [ ] No PHI values appear in log output (log IDs and events, not content)
+- [ ] Audit logs are immutable and stored separately from application logs
+- [ ] Log retention meets the 6-year HIPAA requirement
```
[the remark this answers is not in the document]
Comment (gnpthbalaji) [reply skills/hipaa-compliance/SKILL.md]: Updated the log retention to follow org policies and state laws

**Comment by affaan-m** [comment]
Closing as superseded by direct port. The useful discoverability layer is now on main as a thin  entrypoint that points into the canonical  /  lane instead of introducing a second overlapping healthcare compliance system.

**Comment by affaan-m** [comment]
`hipaa-compliance` is already on `main` as a direct-port adaptation that stays thin and canonical on top of the existing healthcare/privacy lane. Closing this PR as superseded by the direct mainline port.
